leftwm-core/src/handlers/command_handler.rs

```diff
@@ -275,6 +276,71 @@ fn goto_tag(state: &mut State, input_tag: TagId, current_tag_swap: bool) -> Opti
     state.goto_tag_handler(destination_tag)
 }
 
+fn focus_window(state: &mut State, window_name: &str) -> Option<bool> {
+    let is_target = |w: &Window| -> bool {
+        w.res_name
+            .as_ref()
+            .zip(w.res_class.as_ref())
+            .map_or(false, |(res_name, res_class)| {
+                window_name == res_name || window_name == res_class
+            })
+    };
+
+    let current_window = state.focus_manager.window(&state.windows)?;
+    let target_window = if is_target(current_window) {
+        let previous_window_handle = state.focus_manager.window_history.get(1);
+        state
+            .windows
+            .iter()
+            .find(|w| Some(&Some(w.handle)) == previous_window_handle)
+            .cloned()
+    } else {
+        state.windows.iter().find(|w| is_target(*w)).cloned()
+    }?;
+
+    let focus = |s: &mut State, h: WindowHandle| -> Option<bool> { Some(handle_focus(s, h)) };
+    let handle = target_window.handle;
+
+    if target_window.visible() {
+        return focus(state, handle);
+    }
+
+    let tag_id = target_window.tags.first()?;
+    state.goto_tag_handler(*tag_id)?;
+
```

Comment:
After this point we should always return `Some(true)` as we have moved tags so windows need to be redrawn.